internal/app/machined/pkg/system/services/kubelet.go

```diff
@@ -112,9 +114,25 @@ func (k *Kubelet) PreFunc(ctx context.Context, r runtime.Runtime) error {
 	// Pull the image and unpack it.
 	containerdctx := namespaces.WithNamespace(ctx, "k8s.io")
 
-	_, err = image.Pull(containerdctx, r.Config().Machine().Registries(), client, r.Config().Machine().Kubelet().Image())
+	var pullRequired bool
+
+	img, err := client.GetImage(containerdctx, r.Config().Machine().Kubelet().Image())
 	if err != nil {
-		return err
+		pullRequired = true
+	} else {
+		var unpacked bool
+
+		unpacked, err = img.IsUnpacked(containerdctx, "")
+		if err != nil || !unpacked {
+			pullRequired = true
+		}
+	}
+
+	if pullRequired {
+		_, err = image.Pull(containerdctx, r.Config().Machine().Registries(), client, r.Config().Machine().Kubelet().Image())
+		if err != nil {
+			return err
+		}
```

Comment:
Worth putting this in a common place? Like our `containerd` package.

Reply:
yep, I think that makes sense. probably even `image.Pull` function itself with an option?

Reply:
Makes sense.